src/compiler/checker.ts

```diff
@@ -48423,6 +48423,31 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         let current: Node = node;
         while (current) {
             if (isFunctionLikeOrClassStaticBlockDeclaration(current)) {
+                if (node.label) {
+                    const diagnostic = createDiagnosticForNode(node, Diagnostics.Label_0_used_before_declaration, node.label.escapedText.toString());
+
+                    const functionOrClassLike = current as FunctionLikeDeclaration | ClassStaticBlockDeclaration;
+                    if (!functionOrClassLike.body) {
+                        break;
+                    }
+
+                    let matchingLabelDeclaration = false;
+                    forEachChild(functionOrClassLike.body, (childNode) => {
+                        if (matchingLabelDeclaration) {
+                            return;
+                        }

```

Comment:
`forEachChild` bails out early if you return something truthy. Just return `true` as soon as you've found the label.